Networking/Extensions/DecodedExtension.swift

```diff
@@ -29,15 +29,21 @@ public extension Decoded {
     
 }
 
-public extension Decodable where Self: RawRepresentable, Self.RawValue == String {
+public extension Decodable where Self: RawRepresentable, Self.RawValue: Decodable {
     
     static func decode(_ json: JSON) -> Decoded<Self> {
         switch json {
-        case let .string(name): return .fromOptional(Self(rawValue: name))
-        default: return .failure(Argo.DecodeError.custom("Invalid enum value"))
+        case let .string(name) where name is Self.RawValue: return castValueToEnum(name)
+        case let .bool(value) where value is Self.RawValue: return castValueToEnum(value)
+        case let .number(value) where value is Self.RawValue: return castValueToEnum(value)
+        default: return .failure(Argo.DecodeError.custom("Invalid \(Self.self) enum value"))
         }
     }
     
+    private static func castValueToEnum(_ value: Any) -> Decoded<Self> {
```

Comment:
I think you can do `case .string(let name as Self.RawValue): ` and it will only enter if name is of Self.RawValue type. Moreover, can't you remove `Self.` ?